bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -276,20 +274,93 @@ private void manageCurPosIfEnvContainsTarget(final SPTarget target, final Runnab
+        final SPNodeKey newNodeKey = ImOption.apply(getNode()).map(ISPObsComponent::getNodeKey).getOrNull();
+        if (oldNodeKey == null || newNodeKey == null) {
+            return false;
+        }
+        if (!oldNodeKey.equals(newNodeKey)) {
+            return false;
+        }
+
+        // Extract the target environments and check if the auto groups have changed.
+        final TargetEnvironment oldEnv    = oldTOC.getTargetEnvironment();
+        final TargetEnvironment newEnv    = newTOC.getTargetEnvironment();
+        final Option<GuideGroup> oldGpOpt = oldEnv.getGroups().headOption().filter(GuideGroup::isAutomatic);
+        final Option<GuideGroup> newGpOpt = newEnv.getGroups().headOption().filter(GuideGroup::isAutomatic);
+
+        // We only want to return false if the two groups exist and contain the same targets.
+        return oldGpOpt.forall(oldGp ->
+                newGpOpt.forall(newGp -> {
+                    // Extract the target names.
+                    final List<String> oldNames = oldGp.getTargets().map(SPTarget::getName).toList();
+                    final List<String> newNames = newGp.getTargets().map(SPTarget::getName).toList();
+                    return !(oldNames.size() == newNames.size() && oldNames.containsAll(newNames));
+                })
+        );
     }
 
-    @Override public void init() {
+    @Override public void init(final ISPObsComponent oldNode, final TargetObsComp oldTOC) {
+        final boolean autoGroupChanged = autoGroupChanged(oldNode, oldTOC);
+        final TargetObsComp newTOC = getDataObject();
+
+        // If only the auto group has changed, we do not want to use a completely reset target environment, as this
+        // will contain clones of all targets and thus will require the target detail editors to be reset to use these
+        // new clones: this is bad as it will reformat the RA and Dec fields and thus any modifications being made
+        // to them will be lost as they are repopulated and the caret moved to the end.
+        if (autoGroupChanged) {
+            // Determine if there was an auto group in the old env and what the primary index was.
+            final TargetEnvironment oldEnv = oldTOC.getTargetEnvironment();
+            final GuideEnvironment oldGuideEnv = oldEnv.getGuideEnvironment();
+
+            final int primaryIdx               = oldGuideEnv.getPrimaryIndex();
+            final Option<GuideGroup> autoGroup = newTOC.getTargetEnvironment().getGroups().find(GuideGroup::isAutomatic);
+
+            final TargetEnvironment newEnv = autoGroup.map(a -> {
+                final ImList<GuideGroup> newGroups = oldGuideEnv.getOptions().filter(GuideGroup::isManual).cons(a);
+                final GuideEnvironment newGuideEnv = oldGuideEnv.setOptions(newGroups).setPrimaryIndex(primaryIdx);
+                return oldEnv.setGuideEnvironment(newGuideEnv);
+            }).getOrElse(oldEnv);
+            newTOC.setTargetEnvironment(newEnv);
```

Comment:
A fundamental assumption is that there is always an automatic group and that it always be the first group.  This is enforced in the underlying Scala model, though the old Java API doesn't enforce it.  At any rate, I think it'd be okay to just use `GuideEnvironment.setGroup(0, autoGroup)`.  In fact, maybe we could add a `setAutoGroup` to make it more explicit.